### PR TITLE
Issue #SB-16517 feat: api integration for resource component for content

### PR DESCRIPTION
**Pull request by santoshilimi**

### CODE REVIEW

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import { IImpressionEventInput } from '@sunbird/telemetry';
+
+@Component({
+  selector: 'app-contentplayer-page',
+  templateUrl: './contentplayer-page.component.html',
+  styleUrls: ['./contentplayer-page.component.scss']
+})
+export class ContentPlayerPageComponent implements OnInit, OnDestroy {
+  public unsubscribe$ = new Subject<void>();
+  contentId: any;
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts]: This type should be specific not **any**

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  contentDetails: any;
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts]: This also should be interface not **any**

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  playerConfig: PlayerConfig;
+  public telemetryImpression: IImpressionEventInput;
+  constructor(public activatedRoute: ActivatedRoute,
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts]: make the instances private which are not used in the template

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      .subscribe(params => {
+        if (params.contentId) {
+          this.contentId = params.contentId;
+          this.getContent();
+        }
+      });
+  }
+  getContent() {
+    const params = { params: this.configService.appConfig.PublicPlayer.contentApiQueryParams };
+    this.playerService.getContent(this.contentId, params)
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts]: What happens if content get API fails ?

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        this.getContentConfigDetails();
+        this.setTelemetryData();
+      });
+  }
+  getContentConfigDetails() {
+    const contentDetails = {
+      contentId: this.contentId,
+      contentData: this.contentDetails
+    };
+    this.playerConfig = this.playerService.getConfig(contentDetails);
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts]: why player config is set here ?

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  }
+  setTelemetryData() {
```
Comment (harishgilimi) [inline src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts]: This should be called irrespective of content get call success of failed

File: src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts (hunk 5 above)
Comment (santoshilimi) [reply src/app/client/projects/desktop/src/app/modules/content-consumption/components/contentplayer-page/contentplayer-page.component.ts]: Player config data  passing to content player component

**Pull request merged**